### ISSUE TITLE
How to map power button to exit screen?

### ISSUE THREAD

**Issue by ejuarezg** [elenapan/dotfiles#104]
Hi, do you know how I can map the power button on my laptop to the exit screen? I've tried adding the following code to `keys.lua` but it doesn't work

```lua
    awful.key({ }, "XF86PowerOff",
        function ()
            exit_screen_show()
        end,
        {description = "quit awesome", group = "awesome"}),
    awful.key({ }, "XF86PowerDown",
        function ()
            exit_screen_show()
        end,
        {description = "quit awesome", group = "awesome"}),
```

Thanks!

**Comment by elenapan** [comment]
Hi there,

If what you did does not work, then it is probably because `systemd` is handling the power button actions.

This is the shell command which activates the exit screen (you can run it in your terminal to test if it works):
```shell
awesome-client 'exit_screen_show()'
```

So, I think all you need to do is to set the above script to run whenever you press the power button.
I have not tried it, but the related information on how to do this should be [in this Arch Wiki article](https://wiki.archlinux.org/index.php/Power_management#Power_management_with_systemd).

**Comment by ejuarezg** [comment]
The article you linked was helpful, thank you. If anyone else out there has a Macbook Pro 2015 and wants to do this, these are the steps to follow:

1. Change the line `#HandlePowerKey=poweroff` in `/etc/systemd/logind.conf ` to `HandlePowerKey=ignore`
1. Restart `systemd-logind` with `sudo systemctl kill -s HUP systemd-logind`
1. Add the lines
```lua
    awful.key({ }, "XF86PowerOff",
        function ()
            exit_screen_show()
        end,
        {description = "quit awesome", group = "awesome"}),
```
to `keys.lua`
1. Restart awesome

**Comment by elenapan** [comment]
Just tried this on my Lenovo Y50.

Setting `HandlePowerKey='awesome-client "exit_screen_show()"'` did not work for me either so I followed your steps and it worked perfectly.

I have added the `XF86PowerOff` keybind to `keys.lua` (commit baabbe9) and added [a section about this in the Advanced Setup](https://github.com/elenapan/dotfiles/wiki/Advanced-setup#show-exit-screen-when-pressing-the-power-key) wiki page.

Thank you for posting your solution.
